src/dmd/astbase.d

```diff
@@ -4073,7 +4073,11 @@ struct ASTBase
                         e = new ArrayExp(loc, e, cast(Expression)id);
                         break;
 
-                    default:
+                    case DYNCAST.object:
+                    case DYNCAST.tuple:
+                    case DYNCAST.parameter:
+                    case DYNCAST.statement:
+                    case DYNCAST.condition:
```

Comment:
I don't see the value in adding these cases just to `assert(0)`. The default does that just fine.